### ISSUE TITLE
possible to load assets from outside of the assets dir?

### ISSUE THREAD

**Issue by jamesdixon** [lesnitsky/flutter_local_assets_server#4]
Is it possible to serve local assets that have been saved to the app's temp directory rather than through the asset bundle?

Here's the scenario: basically, I'm trying to take advantage of an existing Javascript library to be able to annotate images taken through the app. I basically need to be able to take a photo in the app and then "inject" that into a webview so I can utilize this javascript library to make annotations. 

Would this be possible with your package?

Thank you in advance 🙌

**Comment by sagreine** [comment]
@jamesdixon , did you resolve this? My use case that led me here is pretty different, but I ended up doing something along the lines of:

1. create virtual directory 
2. bind HTTP server to a local address, and on listen serve a file (in my case, a video that the user recorded)
3. send that HTTP request somewhere, e.g. a browser or webview, which is then served the video file.

something like this, though this is typed from memory and bare minimum to get something up, very messy etc...


```
Future<String> _server(String _targetFile) async {
  VirtualDirectory staticFiles = VirtualDirectory('.')
    ..followLinks = true
    ..allowDirectoryListing = true
    ..jailRoot = false;
  var serverRequests =
      // this works if you only want to use your mobile phone to access via browser (navigate to localhost:4040 in a browser on your phone or in the app's flutter webview), but
      // i wanted to be able to access from another pc on network and that didn't work
      //await HttpServer.bind(InternetAddress.loopbackIPv4, 4040);
      await HttpServer.bind('0.0.0.0', 4040);
serverRequests.listen((event) async {
File targetFile = File(_targetFile);
// i think as-coded awaiting here is critical, but obviously you wouldn't do it this way.
assert(await targetFile.exists());
staticFiles.serveFile(targetFile, event);
});
//later, in a function call...
final pickedFile = await picker.getVideo(source: ImageSource.camera);
await _server(pickedFile.path);

```
from there, i could navigate to the phones IP address : /4040 e.g. http://10.0.0.76:4040/ in the browser on a pc and it would serve me the video file that the user just recorded. I didn't personally have a need to go further than that, and haven't messed with non-asset webviews myself.

there is this snippet if you need access to the server programmatically, though you can just use localhost:4040 or whatever port if you're using the phone to access it. 
`String local ="http://" +
      (await GetIp.ipAddress) +
      ":" +
      serverRequests.port.toString();`

No idea if any of that's helpful or not but figured I'd share.

**Comment by hemmrich** [comment]
I'd also love an update on this. I'm trying to download assets (incl HTML files) dynamically and display them in a webview. However, I'm unable to get local_assets_server to use `getapplicationDocumentsDirectory()` as an assetsBasePath.

**Comment by lesnitsky** [comment]
serving assets from outside of app bundle is not supported by this package.
I'll have a look and add this functionality

**Comment by lesnitsky** [comment]
implemented in `v1.0.3`

**Comment by bharathwajv** [comment]
> I'd also love an update on this. I'm trying to download assets (incl HTML files) dynamically and display them in a webview. However, I'm unable to get local_assets_server to use `getapplicationDocumentsDirectory()` as an assetsBasePath.

any luck on implementation?
